### PR TITLE
Hmpid run number in pedestals calculation

### PR DESCRIPTION
**Pull request by fapfap69**

### CODE REVIEW

File: Detectors/HMPID/workflow/src/PedestalsCalculationSpec.cxx [hunk 1]
```diff
@@ -41,6 +41,9 @@
 #include "Framework/Task.h"
 #include "Framework/WorkflowSpec.h"
 #include "Framework/Logger.h"
+#include "Framework/RawDeviceService.h"
+#include "Framework/CommonServices.h"
+#include <fairmq/Device.h>
```
Comment (davidrohr) [inline Detectors/HMPID/workflow/src/PedestalsCalculationSpec.cxx]: Do we need all the includes? Isn't RawDeviceService enough?

**Comment by fapfap69** [comment]
> Il giorno 8 apr 2022, alle ore 17:15, David Rohr ***@***.***> ha scritto:
> 
> 
> @davidrohr commented on this pull request.
> 
> In Detectors/HMPID/workflow/src/PedestalsCalculationSpec.cxx <https://github.com/AliceO2Group/AliceO2/pull/8520#discussion_r846228101>:
> 
> > +#include "Framework/RawDeviceService.h"
> +#include "Framework/CommonServices.h"
> +#include <fairmq/Device.h>
> Do we need all the includes? Isn't RawDeviceService enough?
> 
> 
I did with only RawDeviceService.h, but the compilation ask for fairmq/Device.h

Could be that CommonServices.h I don't need


Cheers

Antonio

File: Detectors/HMPID/workflow/src/PedestalsCalculationSpec.cxx (hunk 1 above)
Comment (fapfap69) [reply Detectors/HMPID/workflow/src/PedestalsCalculationSpec.cxx]: I removed the CommonService.h include

**Comment by fapfap69** [comment]
@davidrohr 
I would test the Store of Pedestals into the CCDB.
Please David, could you provide to merge this PR, thank you.

File: Detectors/HMPID/workflow/src/PedestalsCalculationSpec.cxx [hunk 2]
```diff
@@ -245,13 +252,13 @@ void PedestalsCalculationTask::recordPedInDcsCcdb()
   long maxTimeStamp = minTimeStamp + (3600L * mDcsCcdbAliveHours * 1000);
 
   char filename[1024];
-  sprintf(filename, "%s/%s/PedThre.root", mPedestalsCCDBBasePath.c_str(), mPedestalTag.c_str());
-  mExTimer.logMes("File name = >" + std::string(filename) + "< (" + mPedestalsCCDBBasePath + "," + mPedestalTag);
+  sprintf(filename, "%s/%s/PedThre.root", mPedestalsCCDBBasePath.c_str(), PedestalFixedTag.c_str());
+  mExTimer.logMes("File name = >" + std::string(filename) + "< (" + mPedestalsCCDBBasePath + "," + PedestalFixedTag);
```
Comment (shahor02) [inline Detectors/HMPID/workflow/src/PedestalsCalculationSpec.cxx]: Actually, using the sprintf is dangerous here, you don't know a priory what is the length of the path.
I would use
`auto filename = fmt::format("{}/{}/PedThre.root",mPedestalsCCDBBasePath, PedestalFixedTag);`

**Review by shahor02 — COMMENTED** [review]
@fapfap69 what is the purpose of the file written at endOfStream? Is it written on the EPN (which should not be done) or on the aggregator node? If it is supposed to be persistent (migrated to EOS, it must be accompanied with a special metadata file)

**Comment by shahor02** [comment]
@fapfap69 please rebase your branch to dev and push again, you picked lot of other commits.

**Comment by fapfap69** [comment]
Done ...

**Comment by shahor02** [comment]
I still see a lot of other commits . Please make sure that your dev is up to data then rebase again
```
git checkout dev
git pull
git checkout HMPID-RunNumber_inPedestalsCalculation
git rebase dev
```

**Comment by fapfap69** [comment]
Ok I did.

But, is the third time that I repeted the same actions ...
Could be that a new clean branch is a good strategy 

Antonio

**Comment by shahor02** [comment]
Indeed. Could you make sure that your ~/.gitconfig has lines 
```
[pull]
        rebase = true
```
If not, you can set it by `git config --global pull.rebase true`.
The I would suggest rectifying your dev by `git fetch; git reset --hard origin/dev`, then to create a new branch from dev and to cherry-pick from this branch the commits you need.

**Comment by fapfap69** [comment]
Hi Ruben

I created a new branch and did a new PR ...

Please, clean the previous PR.

Thanks a lot

Antonio

**Pull request closed without merging**